test/cpp_extensions/setup.py

```diff
@@ -27,6 +27,22 @@
         extra_compile_args={'cxx': CXX_FLAGS,
                             'nvcc': ['-O2']})
     ext_modules.append(extension)
+elif torch.cuda.is_available() and ROCM_HOME is not None:
+    from torch.utils.hipify import hipify_python
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    hipify_python.hipify(
+        project_directory=this_dir,
+        output_directory=this_dir,
+        includes="./*",
+        show_detailed=True,
+        is_pytorch_extension=True,)
+    extension = CUDAExtension(
```

Comment:
@ashishfarmer @t-vi do you know what our philosophy is around ROCm vs CUDA? Here I would prefer this to be called "ROCMExtension" (instead of CUDAExtension) so that there is never any ambiguity. But if the rest of pytorch implements ROCM through the CUDA API (i.e., is_cuda() returns True for a tensor on a ROCM device) then this seems OK to me.

Reply:
Yes, currently ROCm masquerades as CUDA for all PyTorch API purposes, so leaving this as CUDAExtension would seem consistent with what is done elsewhere. And really, if we ever make CUDA and ROCm available in the same PyTorch build, we'd probably want to make it a "GPUExtension" (named CUDAExtension, likely) and provide a list of backends this is compiled for.